Fix display

NicolasVereenoghe opened the pull request:
before : http://a-lyrae.com/demo/flump0.2/
aflter : http://a-lyrae.com/demo/flump0.2fix/

Movie::renderFrame partialy rewritten to treat flipbook, spriteSymbol and Movie differently
- scale applied on layer's child
- layer's child x & y modification for Movie
- layer's child pivot.x & pivot.y modification for Sprite & flipbook container
- take into account the offset between transform point and origin in flash

NicolasVereenoghe committed:
fix scale & skew issue
1) scale shifted from the layer to the child (due to how pixi treats scale and skew)
2) offset x & y on the layer's child to reflect the change from container to child scale shifting

NicolasVereenoghe committed:
fix display
Movie::renderFrame algo rewritten to treat flipbook, spriteSymbol and Movie differently
scale applied on layer's child
layer's child x & y modification for Movie
layer's child pivot.x & pivot.y modification for Sprite & flipbook container
take into account the offset between transform point and origin in flash

jackwlee01 commented:
Fantastic! Thanks for looking into this.

@mathieuanthoine could you please integrate this pull request? Thanks!

mathieuanthoine commented:
I'm not sure to have the time right now but as soon as possible.

NicolasVereenoghe commented:
Hello, I would test & adjust something, maybe tonight :

This fix implies that x & y properties of a Movie into a layer are refeshed. It can be annoying, ie responsive design. A friend told me in Flambe there's the same issue, if you want to move a layer, you have to find a way to stop the refresh or clone it to move it.

I realized that instead of updating the x & y properties, changing the pivot seems to render the same thing. I'll try this one, if it works it releases the x & y properties (so that you can update them freely ?)

NicolasVereenoghe commented:
sorry, still some issues, I fear x & y should be refreshed, and by the way, I have found other bad cases :s I'm working on solving them ...

NicolasVereenoghe commented:
ok, new fix again :))

There was a mistake with my last fix, it now manages more cases :

before : http://a-lyrae.com/demo/flump0.2/
aflter : http://a-lyrae.com/demo/flump0.2fix/

I must clear this branch I guess, I'll request a new one for this fix ... I must commit first ...

NicolasVereenoghe committed:
fix the application of pivot on movie container

NicolasVereenoghe commented:
ok, the last fix is commited on this branch ;)

NicolasVereenoghe committed:
Movie::renderFrame again rewritten to fix "resolution" effect on
pivot / scale / x y

new user friendly feature : you can update a movie when no render
refresh is needed (like Flash)

the render of the movie is done to not update tint / pivot / scale / x y
when the MoviePlayer::position of its conatiner doesn't change (stopped)
or for single frame Movie. The movie children still receive
the "render" message for the current "position" so they still play

NicolasVereenoghe commented:
new fix for this branch ;) maybe the last :)
